bundles/org.openhab.core/src/test/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunctionTest.java

```diff
@@ -380,4 +385,160 @@ private GroupItem createGroupItem(String name, Class<? extends Quantity<?>> dime
+
+        Item baseItem = new NumberItem("Number:Temperature", "BaseItem", kelvinProvider);
+        GroupFunction function = new QuantityTypeArithmeticGroupFunction.Avg(Temperature.class, baseItem);
+        State state = function.calculate(items);
+
+        assertEquals(new QuantityType<>("55.33333333333333333333333333333334 °C"), state);
+    }
+
+    @ParameterizedTest
+    @MethodSource("locales")
+    public void testMaxFunctionQuantityTypeDifferentUnitsWithBaseItem(Locale locale) {
+        Locale.setDefault(locale);
+
+        Set<Item> items = new LinkedHashSet<>();
+        items.add(createNumberItem("TestItem1", Temperature.class, new QuantityType<>("100 °C")));
+        items.add(createNumberItem("TestItem2", Temperature.class, UnDefType.NULL));
+        items.add(createNumberItem("TestItem3", Temperature.class, new QuantityType<>("113 °F")));
+        items.add(createNumberItem("TestItem4", Temperature.class, UnDefType.UNDEF));
+        items.add(createNumberItem("TestItem5", Temperature.class, new QuantityType<>("294.15 K")));
+
+        Item baseItem = new NumberItem("Number:Temperature", "BaseItem", kelvinProvider);
+        GroupFunction function = new QuantityTypeArithmeticGroupFunction.Max(Temperature.class, baseItem);
+        State state = function.calculate(items);
+
+        assertEquals(new QuantityType<>("100 °C"), state);
+    }
+
+    @ParameterizedTest
+    @MethodSource("locales")
+    public void testMinFunctionQuantityTypeDifferentUnitsWithBaseItem(Locale locale) {
+        Locale.setDefault(locale);
+
+        Set<Item> items = new LinkedHashSet<>();
+        items.add(createNumberItem("TestItem1", Temperature.class, new QuantityType<>("100 °C")));
+        items.add(createNumberItem("TestItem2", Temperature.class, UnDefType.NULL));
+        items.add(createNumberItem("TestItem3", Temperature.class, new QuantityType<>("113 °F")));
+        items.add(createNumberItem("TestItem4", Temperature.class, UnDefType.UNDEF));
+        items.add(createNumberItem("TestItem5", Temperature.class, new QuantityType<>("294.15 K")));
+
+        Item baseItem = new NumberItem("Number:Temperature", "BaseItem", kelvinProvider);
+        GroupFunction function = new QuantityTypeArithmeticGroupFunction.Min(Temperature.class, baseItem);
+        State state = function.calculate(items);
+
+        assertEquals(new QuantityType<>("294.15 K"), state);
+    }
+
+    @ParameterizedTest
+    @MethodSource("locales")
+    public void testSumFunctionColorTemperatureDifferentUnitsWithBaseItem(Locale locale) {
+        Locale.setDefault(locale);
+
+        Set<Item> items = new LinkedHashSet<>();
+        items.add(createNumberItem("TestItem1", Temperature.class, QuantityType.valueOf(2000, Units.KELVIN)));
+        items.add(createNumberItem("TestItem2", Temperature.class, UnDefType.NULL));
+        items.add(createNumberItem("TestItem3", Temperature.class, QuantityType.valueOf(1726.85, SIUnits.CELSIUS)));
```

Comment:
Is color temperature really ever expressed in °C or °F? I know it now works, but I would just limit it to Kelvin and Mired.

Reply:
I would rather keep the test. I know it should not be commonly used. But it addresses the common case where a user creates a plain vanilla `Number:Temperature` Item without specifying a unit. By default Number:Temperature items have unit=Fahrenheit in USA and unit=Celsius everywhere else. So we need to test that this won't fall over..

Reply:
Yes, I understand. I am still worried that fixing that issue actually makes things worse. Right now the conversion from °C to mirek fails in the conversion as it correctly says it is not the same scale. With the changes, it will succeed. But looking at all of the discussion in the community post, that same user also assumes the state description min and max are in Kelvin when he doesn’t set the unit. Without the error, he would not have known.

Reply:
I think you are referring to #4432